cni/pkg/taint/taintcontroller_test.go

```diff
@@ -0,0 +1,498 @@
+			Status:            v1.ConditionTrue,
+			LastHeartbeatTime: metav1.Time{Time: time.Unix(1, 1)},
+		}
+	} else {
+		nodeReadiness = v1.NodeCondition{
+			Type:              v1.NodeReady,
+			Status:            v1.ConditionFalse,
+			LastHeartbeatTime: metav1.Time{Time: time.Unix(1, 1)},
+		}
+	}
+	return makeNodeWithTaint(makeNodeArgs{NodeName: nodeArgs.nodeName, Taints: []v1.Taint{taint}, NodeCondition: []v1.NodeCondition{nodeReadiness}})
+}
+func TestController_ListAllNode(t *testing.T) {
+	type fields struct {
+		client kubernetes.Interface
+	}
+	type args struct {
+		nodeNames []string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   map[string]bool
+	}{
+		{
+			name: "add a node",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{}, []v1.Node{testingNode}, []v1.ConfigMap{})},
+			args: args{nodeNames: []string{"a"}},
+			want: map[string]bool{"a": true},
+		},
+		{
+			name: "add several node",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{}, []v1.Node{testingNode}, []v1.ConfigMap{})},
+			args: args{nodeNames: []string{"a", "b", "c"}},
+			want: map[string]bool{"a": true, "b": true, "c": true},
+		},
+	}
+	for _, tt := range tests {
+		ts := TaintSetter{configs: []ConfigSettings{}, Client: tt.fields.client}
+		source := fcache.NewFakeControllerSource()
+		tc, _, err := newMockTaintSetterController(&ts, source)
+		if err != nil {
+			t.Fatalf("cannot constrict taint controller")
+		}
+		stop := make(chan struct{})
+		go tc.Run(stop)
+		for _, name := range tt.args.nodeNames {
+			source.Add(&v1.Node{ObjectMeta: metav1.ObjectMeta{Name: name}})
+		}
+		// Let's wait for the controller to finish processing the things we just added.
+		time.Sleep(100 * time.Millisecond)
+		if len(tc.ListAllNode()) != len(tt.args.nodeNames) {
+			t.Fatalf("found %v nodes, expected to have %v nodes", len(tt.args.nodeNames), len(tc.ListAllNode()))
+		}
+		gotItem := make(map[string]bool)
+		for _, node := range tc.ListAllNode() {
+			gotItem[node.Name] = true
+		}
+		if !reflect.DeepEqual(tt.want, gotItem) {
+			t.Fatalf("expected to have %v , found %v", tt.want, gotItem)
+		}
+		close(stop)
+	}
+}
+func TestController_RegistTaints(t *testing.T) {
+	type fields struct {
+		client kubernetes.Interface
+	}
+	type args struct {
+		nodeNames []string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   map[string]bool
+	}{
+		{
+			name: "add a node",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{}, []v1.Node{}, []v1.ConfigMap{})},
+			args: args{nodeNames: []string{"a"}},
+			want: map[string]bool{"a": true},
+		},
+		{
+			name: "add several node",
+			fields: fields{
+				client: fakeClientset([]v1.Pod{}, []v1.Node{}, []v1.ConfigMap{})},
+			args: args{nodeNames: []string{"a", "b", "c"}},
+			want: map[string]bool{"a": true, "b": true, "c": true},
+		},
+	}
+	for _, tt := range tests {
+		ts := TaintSetter{configs: []ConfigSettings{}, Client: tt.fields.client}
```

Comment:
See comment above, use t.Run(). Same applies to all tests, so I won't comment again